test/python/primitives/test_backend_estimator_v2.py

```diff
@@ -0,0 +1,434 @@
+from qiskit.providers.fake_provider import Fake7QPulseV1
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+from qiskit.utils import optionals
+
+BACKENDS = [BasicSimulator(), Fake7QPulseV1(), BackendV2Converter(Fake7QPulseV1())]
+
+
+@ddt
+class TestBackendEstimatorV2(QiskitTestCase):
+    """Test Estimator"""
+
+    def setUp(self):
+        super().setUp()
+        self._precision = 5e-3
+        self._rtol = 3e-1
+        self._seed = 12
+        self.ansatz = RealAmplitudes(num_qubits=2, reps=2)
+        self.observable = SparsePauliOp.from_list(
+            [
+                ("II", -1.052373245772859),
+                ("IZ", 0.39793742484318045),
+                ("ZI", -0.39793742484318045),
+                ("ZZ", -0.01128010425623538),
+                ("XX", 0.18093119978423156),
+            ]
+        )
+        self.expvals = -1.0284380963435145, -1.284366511861733
+
+        self.psi = (RealAmplitudes(num_qubits=2, reps=2), RealAmplitudes(num_qubits=2, reps=3))
+        self.params = tuple(psi.parameters for psi in self.psi)
+        self.hamiltonian = (
+            SparsePauliOp.from_list([("II", 1), ("IZ", 2), ("XI", 3)]),
+            SparsePauliOp.from_list([("IZ", 1)]),
+            SparsePauliOp.from_list([("ZI", 1), ("ZZ", 1)]),
+        )
+        self.theta = (
+            [0, 1, 1, 2, 3, 5],
+            [0, 1, 1, 2, 3, 5, 8, 13],
+            [1, 2, 3, 4, 5, 6],
+        )
+
+    @combine(backend=BACKENDS, abelian_grouping=[True, False])
+    def test_estimator_run(self, backend, abelian_grouping):
+        """Test Estimator.run()"""
+        psi1, psi2 = self.psi
+        hamiltonian1, hamiltonian2, hamiltonian3 = self.hamiltonian
+        theta1, theta2, theta3 = self.theta
+        pm = generate_preset_pass_manager(optimization_level=0, backend=backend)
+        psi1, psi2 = pm.run([psi1, psi2])
+        estimator = BackendEstimatorV2(
+            backend=backend,
+            default_precision=self._precision,
+            options={"abelian_grouping": abelian_grouping, "seed_simulator": self._seed},
+        )
+
+        # Specify the circuit and observable by indices.
+        # calculate [ <psi1(theta1)|H1|psi1(theta1)> ]
+        ham1 = hamiltonian1.apply_layout(psi1.layout)
+        job = estimator.run([(psi1, ham1, [theta1])])
+        result = job.result()
+        np.testing.assert_allclose(result[0].data.evs, [1.5555572817900956], rtol=self._rtol)
+
+        # Objects can be passed instead of indices.
+        # Note that passing objects has an overhead
+        # since the corresponding indices need to be searched.
+        # User can append a circuit and observable.
+        # calculate [ <psi2(theta2)|H1|psi2(theta2)> ]
+        ham1 = hamiltonian1.apply_layout(psi2.layout)
+        result2 = estimator.run([(psi2, ham1, theta2)]).result()
+        np.testing.assert_allclose(result2[0].data.evs, [2.97797666], rtol=self._rtol)
```

Comment:
now you can seed the execution, do you still need `assert_allclose`?

Reply:
Yes. It is necessary because unit tests use both an exact simulator and noise simulators.